### ISSUE TITLE
[App]: Data lost for first several training steps

### ISSUE THREAD

**Issue by YK-YoungK** [wandb/wandb#6744]
### Current Behavior

When I look at curves of training accuracy and validation accuracy, I found something annoying. The data for first 6k steps of training is lost. 
![c23c8dd52a74853a7593dd169571e9f](https://github.com/wandb/wandb/assets/87407123/e950622b-0da7-4f9b-ad68-7696934e1225)
However, when I see the logs, it seems that these data has been uploaded. 
![fe6e0839ed4c065bc1b046e085af365](https://github.com/wandb/wandb/assets/87407123/18cbb6bd-48fc-449b-996a-846ce0612864)
What can I do to recover the curves? I tried again for different experiments but this issue often comes and I don't know how to fix this. Any help will be appreciated.

### Expected Behavior

The training and validation curve should be both complete.

### Screenshots

![c23c8dd52a74853a7593dd169571e9f](https://github.com/wandb/wandb/assets/87407123/26959f68-044f-47d6-9efa-e31d7997c0ef)
![fe6e0839ed4c065bc1b046e085af365](https://github.com/wandb/wandb/assets/87407123/8c77b0b4-88be-4a7b-850e-2117eaa17267)

### Environment

OS: Ubuntu

Browsers: Chrome

Version:

**Comment by YK-YoungK** [comment]
> Hi @YK-YoungK! Thank you for writing in.
> 
> Could you please send a link to your workspace where you are experiencing the first part of your graph not appearing in the UI?

Hi, here is the link: https://wandb.ai/yk-youngk/grok-mlp-default-2?workspace=user-yk-youngk.

**Comment by ArtsiomWB** [comment]
Two quick follow-up questions:
- could you please point me to the graph where you are seeing this concern? not having much luck locating it. 
- Are you logging data every single wandb step?

**Comment by YK-YoungK** [comment]
> Two quick follow-up questions:
> 
> * could you please point me to the graph where you are seeing this concern? not having much luck locating it.
> * Are you logging data every single wandb step?

I see this concern in the figure `train_accuracy, val_accuracy` in project `train_pct=70`. Besides, if the epoch `x` is logged, then the next logging epoch is `max(1.01x, x+1)`.

**Comment by ArtsiomWB** [comment]
Could you please add that figure to a report and then share it with me? Still no luck on my side. 

I think what could be happening is our Sampling behavior. After we have logged >1500 points our chart sampling kicks in for performance reasons. If more than 1500 points are being displayed at once, some of the points get combined to prevent slowing down the UI. To confirm this what you can do is simply zoom in to the part of the graph that is missing in your UI and see if those points start appearing again.

**Comment by ArtsiomWB** [comment]
Hi there, I wanted to follow up on this request. Please let us know if we can be of further assistance or if your issue has been resolved.

**Comment by YK-YoungK** [comment]
> Could you please add that figure to a report and then share it with me? Still no luck on my side.
> 
> I think what could be happening is our Sampling behavior. After we have logged >1500 points our chart sampling kicks in for performance reasons. If more than 1500 points are being displayed at once, some of the points get combined to prevent slowing down the UI. To confirm this what you can do is simply zoom in to the part of the graph that is missing in your UI and see if those points start appearing again.

Hi, here is one report: https://api.wandb.ai/links/yk-youngk/0s97madx. Besides, when I zoom into the part of the missing curve the data actually appears.

**Comment by ArtsiomWB** [comment]
Thank you so much for sending it over! It does look like sampling. Our engineers are aware of the issue and are looking into a fix. Thank you so much for bringing this up!

**Comment by verbose-void** [comment]
same issue here. causing serious chunks of data missing even when upgrading to the latest 0.17.5 version

**Comment by ArtsiomWB** [comment]
@verbose-void, could you point me to the workspace where you are seeing this behavior?

**Comment by ArtsiomWB** [comment]
Hi there, I wanted to follow up on this request. Please let us know if we can be of further assistance or if your issue has been resolved.
